client/src/watcher/mod.rs

```diff
@@ -0,0 +1,80 @@
+mod pool;
+
+use crate::watcher::pool::Pool;
+use anyhow::{anyhow, Result};
+use async_trait::async_trait;
+use log::{debug, info};
+use notify::{DebouncedEvent, RecursiveMode, Watcher};
+use std::sync::{Arc, Mutex};
+use std::time::Duration;
+
+#[async_trait]
+pub trait WatcherListener {
+    /// Function called whenever a file is created
+    async fn on_event(&self, event: &DebouncedEvent) -> Result<()>;
+}
+
+#[derive(Clone)]
+pub struct PoolWatcher {
+    /// The pool of files to watch
+    pub(crate) pool: Pool,
+
+    /// Listener suscribed to the file watch events
+    pub(crate) listeners: Vec<Arc<Mutex<dyn WatcherListener>>>,
+}
+
+impl PoolWatcher {
```

Comment:
Why making everything `async` in this module ? It guess it's not necessary, and having a notification channel would prevent this. `async` is not the same as in javascript, we can't put it everywhere :)